packages/protocol/contracts/tokenvault/ERC20Vault.sol

```diff
@@ -130,10 +131,13 @@ contract ERC20Vault is BaseVault {
 
         btokenOld = canonicalToBridged[ctoken.chainId][ctoken.addr];
 
+        // TODO(Brecht): if the ctoken is on the current chain, should we check the
+        // symbol/name/decimals against the actual token?
```

Comment:
Strangely enough `_handleMessage` gets that data directly from the token contract, but it's not checked here. Any reason?

Reply:
I think changeBridgedToken is called only on the target chain but ctoken is on the remote/source chain.

Reply:
Oh I see, I thought it also used some stored data in that case, but seems not!